src/common/fail_loc.c

```diff
@@ -81,8 +81,12 @@ daos_fail_loc_set(uint64_t fail_loc)
 	else
 		attr_in.fa_id = DAOS_FAIL_GROUP_GET(fail_loc);
 
-	D_ASSERT(attr_in.fa_id > 0);
-	D_ASSERT(attr_in.fa_id < DAOS_FAIL_MAX_GROUP);
+	if ((int)attr_in.fa_id < 0 ||
```

Comment:
why not change the type of fa_id if it can be signed?

Reply:
hmm,  this > 0 check probably should be removed.